### PR TITLE
feat(module:tabs): support link router

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by wendzhue** [comment]
Working on tests... 🤗

**Comment by lppedd** [comment]
@wendzhue Thank you for this one!
I'm using un ugly workaround now, finally I can remove it.

File: components/tabs/nz-tab.component.ts [hunk 1]
```diff
@@ -22,11 +22,30 @@ import {
...
 
 import { InputBoolean } from 'ng-zorro-antd/core';
+import { Subject } from 'rxjs';
+
 import { NzTabDirective } from './nz-tab.directive';
 
+/**
+ * This component is for catching `routerLink` directive.
+ */
+@Component({
```
Comment (vthinkxie) [inline components/tabs/nz-tab.component.ts]: should be a directive here, and move to another file

File: components/tabs/nz-tabset.component.ts [hunk 2]
```diff
@@ -289,5 +306,53 @@ export class NzTabSetComponent
...
+
+  private isLinkActive(router: Router): (link?: RouterLink | RouterLinkWithHref) => boolean {
+    return (link?: RouterLink | RouterLinkWithHref) => (link ? router.isActive(link.urlTree, true) : false);
+  }
+
+  private tryActivateLink(index: number): void {
+    const tabs = this.listOfNzTabComponent.toArray();
+    const routerLink = tabs[index].titleComponent.routerLinkWithHref!;
+
+    console.log(routerLink),
```
Comment (vthinkxie) [inline components/tabs/nz-tabset.component.ts]:
```suggestion
```

File: components/tabs/demo/link-router.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <nz-tabset nzLinkRouter>
+      <nz-tab>
+        <span nz-tab-title
+          ><a [routerLink]="['.']" [queryParams]="{ tab: 'one' }" queryParamsHandling="merge">Default</a></span
+        >
+        Default.
+      </nz-tab>
+      <nz-tab>
+        <span nz-tab-title
+          ><a [routerLink]="['.']" [queryParams]="{ tab: 'two' }" queryParamsHandling="merge">Two</a></span
```
Comment (vthinkxie) [inline components/tabs/demo/link-router.ts]: `<a nz-tab-link [routerLink]="['.']" [queryParams]="{ tab: 'two' }" queryParamsHandling="merge">Two</a>`
is a better API I think

**Pull request merged**